### PR TITLE
Combine all styles

### PR DESCRIPTION
**Pull request by dipinknair**
Combine packaging, coding and documentation styles under Style section

### CODE REVIEW

File: doc/source/all-styles.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
+
+Style
+=====
+
+In the PyAnsys ecosystem, three key styles — Packaging, Coding, and Documentation
```
Comment (PipKat) [inline doc/source/all-styles.rst:5]:
```suggestion
In the PyAnsys ecosystem, three key styles—packaging, coding, and documentation—
```

File: doc/source/all-styles.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
+
+Style
+=====
+
+In the PyAnsys ecosystem, three key styles — Packaging, Coding, and Documentation 
+serve as foundational pillars to enhance the overall developer experience.
+
+The Packaging style focuses on creating clear, open-source APIs hosted on GitHub,
```
Comment (PipKat) [inline doc/source/all-styles.rst:8]:
```suggestion
Packaging style focuses on creating clear, open source APIs hosted on GitHub,
```
Google dev guide says it is "open source" even when used as a compound adjective.

File: doc/source/all-styles.rst [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+allowing for reusable packages that can be updated independently of the Ansys release schedule.
+The Coding style adheres to PEP 8 and aligns with the conventions of 
+major data science libraries like numpy, scipy, and pandas, ensuring consistency and readability.
+Lastly, the Documentation style emphasizes the significance of well-documented APIs, 
+offering increased adoption, improved onboarding experiences, and streamlined code maintenance.
```
Comment (PipKat) [inline doc/source/all-styles.rst:13]:
```suggestion
- Coding style adheres to PEP 8 and aligns with the conventions of major data science libraries
like NumPy, SciPy, and pandas, ensuring consistency and readability.
- Documentation style emphasizes the significance of well-documented APIs, 
offering increased adoption, an improved on-boarding experiences, and streamlined code maintenance.
```

**Comment by dipinknair** [comment]
image

File: doc/source/all-styles.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+- Packaging style focuses on creating clear, open source APIs hosted on GitHub,allowing
+  for reusable packages that can be updated independently of the Ansys release schedule.
+
+- Coding style adheres to PEP 8 and aligns with the conventions of major data science libraries
+  like NumPy, SciPy, and pandas, ensuring consistency and readability.
+
+- Documentation style emphasizes the significance of well-documented APIs, 
+  offering increased adoption, an improved on-boarding experiences, and streamlined code maintenance.
+
+All PyAnsys libraries follow the Google developer documentation style guide,
```
Comment (PipKat) [inline doc/source/all-styles.rst:17]: Make ``Google developer documentation style guide`` a link: https://developers.google.com/style/

File: doc/source/all-styles.rst [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        :link: coding-style/index
+        :link-type: doc
+
+        Best practices for writing Python code.
+
+    .. grid-item-card:: :octicon:`pencil` Documentation
+        :link: doc-style/index
+        :link-type: doc
+
+        Best practices for writing documentation.
```
Comment (PipKat) [inline doc/source/all-styles.rst:40]:
```suggestion
        Best practices for writing PyAnsys library documentation.
```

**Review by PipKat — APPROVED** [review]

File: doc/source/all-styles.rst (hunk 4 above)
Comment (dipinknair) [reply doc/source/all-styles.rst:17]: Did you mean the space for the second line of bullet point? 
?

**Comment by RobPasMue** [comment]
These are breaking changes. Before merging we should inform everybody. Links will start to fail all around our docs.

**Comment by RobPasMue** [comment]
> These are breaking changes. Before merging we should inform everybody. Links will start to fail all around our docs.

My bad - it won't break anything. Each subsection will still be at the same URL since you didn't move the files. You just created a new one. All good from my side!

File: doc/source/all-styles.rst [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+- Packaging style focuses on creating clear, open source APIs hosted on GitHub,allowing
+  for reusable packages that can be updated independently of the Ansys release schedule.
+- Coding style adheres to PEP 8 and aligns with the conventions of major data science libraries
+  like NumPy, SciPy, and pandas, ensuring consistency and readability.
+- Documentation style emphasizes the significance of well-documented APIs,
+  offering increased adoption, an improved on-boarding experiences, and streamlined code maintenance.
+
+All PyAnsys libraries follow the `Google developer documentation style guide
+<https://developers.google.com/style/>`_, incorporating sentence case, 
+active voice, and concise, clear sentences for cohesive and user-friendly documentation.
```
Comment (RobPasMue) [inline doc/source/all-styles.rst:17]: I think this info shouldn't go here, but in the "doc style" section.
```suggestion
```

File: doc/source/all-styles.rst [hunk 7]
```diff
@@ -0,0 +1,46 @@
+
+Style
+=====
+
+In the PyAnsys ecosystem, three key styles—packaging, coding, and documentation—
```
Comment (RobPasMue) [inline doc/source/all-styles.rst:5]:
```suggestion
In the PyAnsys ecosystem, three key styles —packaging, coding, and documentation—
```

**Review by RobPasMue — APPROVED** [review]

**Comment by PipKat** [comment]
> > These are breaking changes. Before merging we should inform everybody. Links will start to fail all around our docs.
> 
> My bad - it won't break anything. Each subsection will still be at the same URL since you didn't move the files. You just created a new one. All good from my side!

@RobPasMue Not that I've had any time to move PyAnsys writer's guide content into this dev guide, but if I did, I feel like I'd need a strategic meeting with you to discuss how best to integrate it. Chris H. suggested reorganizing the dev guide as I do the integration, but I'm sure that would result in broken links.

**Comment by RobPasMue** [comment]
> > > These are breaking changes. Before merging we should inform everybody. Links will start to fail all around our docs.
> > 
> > 
> > My bad - it won't break anything. Each subsection will still be at the same URL since you didn't move the files. You just created a new one. All good from my side!
> 
> @RobPasMue Not that I've had any time to move content from the PyAnsys writer's guide into the dev guide, but if I did, I feel like I'd need a strategic meeting with you to discuss how best to integrate it. Chris H. suggested reorganizing the dev guide as I do the integration, but I'm sure that would result in broken links.

Yep, we should have a meeting prior to that taking place because it would break many people's repos. Let's talk offline! :)

File: doc/source/all-styles.rst (hunk 7 above)
Comment (PipKat) [reply doc/source/all-styles.rst:5]: @RobPasMue @dipinknair Google dev style guideline says no spaces before dashes. (This is also true in other style guides.) I generally prefer using commas over dashes: "In the PyAnsys ecosystem, three key styles, packaging, coding, and documentation, ...." Vale will throw an error if the spaces appear before the dashes.

Comment (RobPasMue) [reply doc/source/all-styles.rst:5]: Oh, didn't know that!

**Pull request merged**